mllib/src/main/scala/org/apache/spark/mllib/clustering/BisectingKMeansModel.scala

```diff
@@ -195,7 +200,7 @@ object BisectingKMeansModel extends Loader[BisectingKMeansModel] {
       val data = rows.select("index", "size", "center", "norm", "cost", "height", "children")
       val nodes = data.rdd.map(Data.apply).collect().map(d => (d.index, d)).toMap
       val rootNode = buildTree(rootId, nodes)
-      new BisectingKMeansModel(rootNode, DistanceMeasure.EUCLIDEAN)
+      new BisectingKMeansModel(rootNode, DistanceMeasure.EUCLIDEAN, 0.0)
```

Comment:
Is it possible to compute the cost here after load rather than setting it to 0?

Reply:
Despite it would be possible, I don't think it is a good idea, as this is only for the old MLLib API and it would introduce a significant performance overhead (a pass over all the dataset) for an information which may not be useful at all...

Reply:
Would it not just be the same? `rootNode.leafNodes.map(_.cost).sum`? If that cost info is present in the nodes (?) it doesn't need a pass over data (which indeed doesn't exist at this point). If it's valuable enough to include at all, should this info not be correct where it is in fact available?

Reply:
yes, you're right, thanks. It is indeed available. I am doing it, thanks.

Reply:
I think it looks OK except for this comment?

Reply:
yes, right, sorry, I missed it, I did it only for the version 2.0 and missed this one. I am updating it, thanks.